src/main/java/com/codeborne/selenide/impl/WebdriverPhotographer.java

```diff
@@ -1,24 +1,94 @@
 package com.codeborne.selenide.impl;
 
 import com.codeborne.selenide.Driver;
+import com.github.bsideup.jabel.Desugar;
 import org.openqa.selenium.OutputType;
 import org.openqa.selenium.TakesScreenshot;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chromium.ChromiumDriver;
+import org.openqa.selenium.devtools.DevTools;
+import org.openqa.selenium.devtools.HasDevTools;
+import org.openqa.selenium.devtools.v101.page.Page;
+import org.openqa.selenium.devtools.v101.page.model.Viewport;
+import org.openqa.selenium.remote.Augmenter;
+import org.openqa.selenium.remote.RemoteWebDriver;
 
 import javax.annotation.CheckReturnValue;
 import javax.annotation.Nonnull;
 import javax.annotation.ParametersAreNonnullByDefault;
+import java.util.HashMap;
+import java.util.Map;
 import java.util.Optional;
 
 @ParametersAreNonnullByDefault
 public class WebdriverPhotographer implements Photographer {
+  private static final JavaScript js = new JavaScript("get-screen-size.js");
+
   @Nonnull
   @CheckReturnValue
   @Override
   public <T> Optional<T> takeScreenshot(Driver driver, OutputType<T> outputType) {
+    if (driver.getWebDriver() instanceof ChromiumDriver chromiumDriver) {
```

Comment:
You cannot rely on such type guard. Webdriver instance can be decorated e.g. from webdriver listeners.
example:
```
public class MainPageTest {
    @Test
    public void test() {
        WebDriverRunner.addListener(new WebDriverListener() {
            @Override
            public void beforeAnyCall(Object target, Method method, Object[] args) {

            }
        });
        Selenide.open();
        WebDriver driver = WebDriverRunner.getWebDriver();
        System.out.println(driver.getClass()); // net.bytebuddy.renamed.java.lang.Object$ByteBuddy$gM4k6yJv
        System.out.println(Arrays.toString(driver.getClass().getInterfaces())); // all ChromeDriver interfaces
        assertTrue(driver instanceof ChromiumDriver); // fail :/
    }
}
```

Reply:
@BorisOsipov Valid point. Fixed in https://github.com/selenide/selenide/pull/1858.